layer/vulkan_reflex_layer.cpp

```diff
@@ -0,0 +1,896 @@
+#include <algorithm>
+#include <cinttypes>
+#include <cstdint>
+#include <cstdlib>
+#include <mutex>
+#include <optional>
+#include <string_view>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+
+#define VK_NO_PROTOTYPES
+
+#include <vulkan/vulkan.h>
+
+// hijack VKAPI_ATTR before vk_layer.h is included to make sure that vkNegotiateLoaderLayerInterfaceVersion
+// has appropriate visibility attribute in its initial declaration / prototype, otherwise MSVC will complain
+
+#undef VKAPI_ATTR
+
+#if defined(_WIN32)
+#define VKAPI_ATTR __declspec(dllexport)
+#elif defined(__GNUC__)
+#define VKAPI_ATTR [[gnu::visibility("default")]]
```

Comment:
As a side note, I would prefer to do the attribute check first because at least that is standard, and then leave the `declspec` for MSVC-like compilers…

```cpp
#if __has_cpp_attribute(gnu::visibility)
#define VKAPI_ATTR [[gnu::visibility("default")]]
#elif defined(_MSC_VER)
#define VKAPI_ATTR __declspec(dllexport)
```

But this is how vkroots does it and unfortunately my version would trip `clang-cl` which supports both, causing it to complain about `dllexport` being added later :slightly_frowning_face:

Maybe the idea of having a Windows-side layer is not worth the effort.